### PR TITLE
Add Metric component

### PR DESCRIPTION
**Pull request by Janpot**
Closes https://github.com/mui/mui-toolpad/issues/2025

https://github.com/mui/mui-toolpad/assets/2109932/c4a98af8-15e3-4139-a558-9d069cef5043

To do:

- [x] "colorScale" => "Conditional format"
- [x] "numberFormat" => "Number format"
- [x] "statistic" => "metric"

Future:

- sparkline
- difference with previous value

### CODE REVIEW

**Comment by apedroferreira** [comment]
Hey, some thoughts from watching the video:
- Why is `numberFormat` in its own popover? Could it not just be a dropdown in the props editor?
- The colors stops popover feels a bit empty in it's initial state, at least on the left side of the colored circle. Maybe there could be some text there to make things more explanatory, not sure exactly what, possibly "Default color"?

File: docs/data/toolpad/reference/components/metric.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!-- This file has been auto-generated using `yarn docs:build:api`. -->
+
+# Metric
+
+<p class="description">API docs for the Toolpad Metric component.</p>
+
+The Metric component lets you display values.
```
Comment (apedroferreira) [inline docs/data/toolpad/reference/components/metric.md:7]: Should we elaborate a bit more on possible typical use cases in this description?

**Comment by Janpot** [comment]
> * Why is numberFormat in its own popover? Could it not just be a dropdown in the props editor?

It could have more inputs depending on the context:

Screenshot 2023-07-12 at 09 22 03

maybe there is a better way to do this

**Review by apedroferreira — APPROVED** [review]
The changes look great, I think they were good choices!

**Pull request merged**